pkg/util/utils.go

```diff
@@ -311,36 +311,37 @@ func GetDefaultStoreOptions() (storage.StoreOptions, error) {
 		storageOpts = storage.StoreOptions{}
 		storage.ReloadConfigurationFile(storageConf, &storageOpts)
 	}
-
-	if rootless.IsRootless() {
-		if os.IsNotExist(err) {
-			os.MkdirAll(filepath.Dir(storageConf), 0755)
-			file, err := os.OpenFile(storageConf, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
-			if err != nil {
-				return storageOpts, errors.Wrapf(err, "cannot open %s", storageConf)
-			}
-
-			tomlConfiguration := getTomlStorage(&storageOpts)
-			defer file.Close()
-			enc := toml.NewEncoder(file)
-			if err := enc.Encode(tomlConfiguration); err != nil {
-				os.Remove(storageConf)
-			}
-		} else if err == nil {
-			// If the file did not specify a graphroot or runroot,
-			// set sane defaults so we don't try and use root-owned
-			// directories
-			if storageOpts.RunRoot == "" {
-				storageOpts.RunRoot = defaultRootlessRunRoot
-			}
-			if storageOpts.GraphRoot == "" {
-				storageOpts.GraphRoot = defaultRootlessGraphRoot
-			}
+	if rootless.IsRootless() && err == nil {
+		// If the file did not specify a graphroot or runroot,
+		// set sane defaults so we don't try and use root-owned
+		// directories
+		if storageOpts.RunRoot == "" {
+			storageOpts.RunRoot = defaultRootlessRunRoot
+		}
+		if storageOpts.GraphRoot == "" {
+			storageOpts.GraphRoot = defaultRootlessGraphRoot
```

Comment:
@giuseppe Probably a matter for a separate PR, but would it be worth also setting overlay and overlay mountprogram in the defaults here? Might fix a lot of our "overlayfs is not supported over this filesystem" errors

Reply:
I can do that, should I hardcode it to "/usr/bin/fuse-overlayfs"?

Reply:
I think that ought to be fine - even if it doesn't exist, we get a much nicer error than the current "overlay is not supported over this filesystem" one

Reply:
hm.. looking at the code that should not really happen, if fuse-overlayfs is not found then it uses vfs